storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -3346,6 +3346,12 @@ public void rebalance(String topoName, RebalanceOptions options)
             if (subject != null) {
                 options.set_principal(subject.getPrincipals().iterator().next().getName());
             }
+            String topoId = toTopoId(topoName);
+            StormTopology stormTopology = tryReadTopology(topoId, topoCache);
+            StormBase base = new StormBase();
+            base.set_name(topoName);
+            idToExecutors.getAndUpdate(new Assoc<>(topoId,
+                    new HashSet<>(computeExecutors(topoId, base, topoConf, stormTopology))));
```

Comment:
When rebalance happens, 
the code here  https://github.com/apache/storm/blob/master/storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java#L1749 will remove the entry and let it recompute. 
So I think we don't this?